### PR TITLE
ci: add Node.js v23

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?
CI

#### What changes did you make? (Give an overview)
Add Node.js v23 environment
#### Related Issues
No
#### Is there anything you'd like reviewers to focus on?
No

### CODE REVIEW

**Review by fasttime — DISMISSED** [review]
LGTM, thanks! But there are other checks failing. We should fix those problems first I think.

**UPDATE**: It seems that the CI build will be fixed when #42 is merged.

**Comment by amareshsm** [comment]
> LGTM, thanks! But there are other checks failing. We should fix those problems first I think.

In local I am not getting errors. I check in the node v20 and 22.

**Comment by amareshsm** [comment]
> > In local I am not getting errors. I check in the node v20 and 22.
> 
> I am getting errors locally after deleting `node_modules` and running `npm install` again and then `npm run build`.

ya, I am also getting errors after deleting and installing `node_modules`.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
         strategy:
             matrix:
                 os: [ubuntu-latest]
-                node: [22.x, 21.x, 20.x, 18.x, "18.18.0"]
+                node: [23.x, 22.x, 21.x, 20.x, 18.x, "18.18.0"]
```
Comment (nzakas) [inline .github/workflows/ci.yml:32]: Let's remove 21.x. Odd numbered versions always supersede previous odd numbered versions.

Comment (snitin315) [reply .github/workflows/ci.yml:32]:
```suggestion
                node: [23.x, 22.x, 20.x, 18.x, "18.18.0"]
```

**Comment by snitin315** [comment]
@fasttime @nzakas Done!

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**